pyaedt/edb_grpc/core/components.py

```diff
@@ -0,0 +1,2209 @@
+                    if pins:
+                        if len(pins) == 1:
+                            pin_term = self._create_terminal(pins[0])
+                            if pin_term:
+                                pin_term.SetReferenceTerminal(ref_pin_group_term)
+                        else:
+                            pin_group = self.create_pingroup_from_pins(pins)
+                            if not pin_group:
+                                return False
+                            pin_group_term = self._create_pin_group_terminal(pin_group, component)
+                            if pin_group_term:
+                                pin_group_term.SetReferenceTerminal(ref_pin_group_term)
+                    else:
+                        self._logger.info("No pins found on component {} for the net {}".format(component, net))
+            else:
+                ref_pin_group = self.create_pingroup_from_pins(ref_pins)
+                if not ref_pin_group:
+                    self._logger.warning("failed to create reference pin group")
+                    return False
+                ref_pin_group_term = self._create_pin_group_terminal(ref_pin_group, component, isref=True)
+                for net in net_list:
+                    pins = [pin for pin in cmp_pins if pin.GetNet().GetName() == net]
+                    for pin in pins:
+                        pin_group = self.create_pingroup_from_pins([pin])
+                        pin_group_term = self._create_pin_group_terminal(pin_group, component, isref=False)
+                        pin_group_term.SetReferenceTerminal(ref_pin_group_term)
+        return True
+
+    @pyaedt_function_handler()
+    def _create_terminal(self, pin):
+        """Create terminal on component pin.
+
+        Parameters
+        ----------
+        pin : Edb padstack instance.
+
+        Returns
+        -------
+        Edb terminal.
+        """
+
+        pin_position = self.get_pin_position(pin)  # pragma no cover
+        pin_pos = self._edb.Geometry.PointData(
+            self._get_edb_value(pin_position[0]), self._get_edb_value(pin_position[1])  # pragma no cover
+        )
+        res, from_layer, _ = pin.GetLayerRange()
+        cmp_name = pin.GetComponent().GetName()
+        net_name = pin.GetNet().GetName()
+        pin_name = pin.GetName()
+        term_name = "{}.{}.{}".format(cmp_name, pin_name, net_name)
+        term = self._edb.Cell.Terminal.PointTerminal.Create(
+            pin.GetLayout(), pin.GetNet(), term_name, pin_pos, from_layer
+        )
+        return term
+
+    @pyaedt_function_handler()
+    def _get_closest_pin_from(self, pin, ref_pinlist):
+        """Returns the closest pin from given pin among the list of reference pins.
+
+        Parameters
+        ----------
+        pin : Edb padstack instance.
+
+        ref_pinlist : list of reference edb pins.
+
+        Returns
+        -------
+        Edb pin.
```

Comment:
```suggestion
        pin : EDB padstack instance.

        ref_pinlist : List of reference EDB pins.

        Returns
        -------
        EDB pin.
```
Again--no data types and relatively weak descriptions.